cannon/mipsevm/tests/evm_multithreaded_test.go

```diff
@@ -1193,6 +1193,37 @@ func TestEVM_UnsupportedSyscall(t *testing.T) {
 	}
 }
 
+func TestEVM_EmptyThreadStacks(t *testing.T) {
+	t.Parallel()
+	var tracer *tracing.Hooks
+
+	cases := []struct {
+		name           string
+		otherStackSize int
+		traverseRight  bool
+	}{
+		{name: "Traverse right with empty stacks", otherStackSize: 0, traverseRight: true},
+		{name: "Traverse left with empty stacks", otherStackSize: 0, traverseRight: false},
+		{name: "Traverse right with one non-empty stack on the other side", otherStackSize: 1, traverseRight: true},
+		{name: "Traverse left with one non-empty stack on the other side", otherStackSize: 1, traverseRight: false},
+	}
+
+	for i, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			goVm, state, contracts := setup(t, i*123, nil)
+			mttestutil.SetupThreads(int64(i*123), state, c.traverseRight, 0, c.otherStackSize)
+
+			// Since the verification of the proof is always performed after the active thread check,
+			// it is sufficient to pass a dummy proof here.
+			dummyProof := multiThreadedProofGenerator(t, multithreaded.CreateEmptyState())
```

Comment:
(optional) We discussed offline adding some variations on the proof data here, but I think this is just a nice-to-have. 

For example, we could generate a few testcases like:
```
func GenerateEmptyThreadProofVariations() ([]threadProofTestcase){
   defaultThreadProof := multiThreadedProofGenerator(t, multithreaded.CreateEmptyState())
   zeroBytesThreadProof := make([]byte, THREAD_WITNESS_SIZE)
   zeroBytesThreadProof[SERIALIZED_THREAD_SIZE:] = defaultThreadProof[SERIALIZED_THREAD_SIZE:]
   nilBytesThreadProof := defaultThreadProof[SERIALIZED_THREAD_SIZE:]
    return []{
        {name: "default thread proof", proof: defaultThreadProof},
        {name: "zeroed thread bytes proof", proof: zeroBytesThreadProof},
        {name: "nil thread bytes proof", proof: nilBytesThreadProof}, 
    }
}
```